kalamine/layout.py

```diff
@@ -104,6 +112,114 @@ def load_tpl(layout: "KeyboardLayout", ext: str) -> str:
+        msklc = msklc_dir / Path("MSKLC.exe")
+        subprocess.run([msklc, klc_file, "-build"], capture_output=not verbose)
+    
+        # move the installer from "My Documents" to current dir
+        CSIDL_PERSONAL = 5  # My Documents
+        SHGFP_TYPE_CURRENT = 0  # Get current, not default value
+        buf = ctypes.create_unicode_buffer(ctypes.wintypes.MAX_PATH)
+        ctypes.windll.shell32.SHGetFolderPathW(
+            None, CSIDL_PERSONAL, None, SHGFP_TYPE_CURRENT, buf
+        )
+        my_docs = Path(buf.value)
+        installer = my_docs / Path(name8)
+    
+        if installer.exists():
+            move(installer, working_dir / Path(name8))
+    
+    
+    def build_msklc_dll(
+        layout: "KeyboardLayout",
+        msklc_dir: Path,
+        verbose: bool,
+        working_dir: Path = Path(os.getcwd()),
+    ):
+        name8 = layout.meta["name8"]
+        prev = os.getcwd()
+        os.chdir(working_dir)
+        INST_DIR = working_dir / Path(name8)
+        dll_dirs = ["i386", "amd64", "ia64", "wow64"]
+        for dll_dir in dll_dirs:
+            full_dir = INST_DIR / Path(dll_dir)
+            if not full_dir.exists():
+                raise Exception(f"{full_dir} doesn't exists")
+            else:
+                rmtree(full_dir)
+                os.mkdir(full_dir)
+    
+        klc_file = working_dir / Path(f"{name8}.klc")
+    
+        # create correct klc
+        with klc_file.open("w", encoding="utf-16le", newline="\n") as file:
+            file.write(layout.klc)
+    
+        create_c_files(layout, msklc_dir, verbose, working_dir)
+        c_file = klc_file.with_suffix(".RC")
+        with c_file.open("w", encoding="utf-16le", newline="\n") as file:
+            file.write(layout.rc)
+        c_file = klc_file.with_suffix(".C")
+        with c_file.open("w", encoding="utf-16le", newline="\n") as file:
+            file.write(layout.c)
+        c_files = [".C", ".RC", ".H", ".DEF"]
+        # Set files to read only to avoid overwrite by buggy MSKLC
+        for suffix in c_files:
+            os.chmod(klc_file.with_suffix(suffix), S_IREAD)
+    
+        # build correct DLLs
+    
+        kbdutools = msklc_dir / Path("bin/i386/kbdutool.exe")
+        dll = klc_file.with_suffix(".dll")
+    
+        subprocess.run(
+            [kbdutools, "-u", "-x", klc_file], capture_output=not verbose
+        ).check_returncode()
+        move(dll, INST_DIR / Path("i386"))
+    
+        subprocess.run(
+            [kbdutools, "-u", "-m", klc_file], capture_output=not verbose
+        ).check_returncode()
+        move(dll, INST_DIR / Path("amd64"))
+    
+        subprocess.run(
+            [kbdutools, "-u", "-i", klc_file], capture_output=not verbose
+        ).check_returncode()
+        move(dll, INST_DIR / Path("ia64"))
+    
+        subprocess.run(
+            [kbdutools, "-u", "-o", klc_file], capture_output=not verbose
+        ).check_returncode()
+        move(dll, INST_DIR / Path("wow64"))
```

Comment:
`capture_output=not verbose` is to avoid having all the errors about being unable to write the C files as we’re hacking into kbdutools to produce the DLL using our C code